clients/src/main/java/org/apache/kafka/common/ClusterResourceListener.java

```diff
@@ -24,7 +24,7 @@
  * <p>
  * <h4>Clients</h4>
  * There will be one invocation of {@link ClusterResourceListener#onUpdate(ClusterResource)} after each metadata response.
- * Note that the cluster id may be null when the Kafka broker version is below 0.10.1.0. If you receive a null cluster id, you can expect it to always be null unless you have a cluster with multiple broker versions which can happen if the cluster is being upgraded while the client is running.
+ * Note the minimum supported broker version is 2.1.
```

Comment:
I think we can remove this line altogether. Best to avoid having versions mentioned in every class.